### PR TITLE
TSK-1565: Make Wildcard Queries Case insensitive

### PR DESCRIPTION
**Pull request by gitgoodjhe**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [x] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [x] Readability

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java [hunk 1]
```diff
@@ -61,6 +61,32 @@ void should_ReturnAllTasks_For_ProvidedSearchValueAndAdditionalParameters() {
     assertThat(foundTasks).hasSize(1);
   }
 
+  @WithAccessId(user = "admin")
+  @Test
+  void should_ReturnAllTasks_For_CaseInsensitiveValue() {
```
Comment (tge20) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java:66]: As I remember our pattern is "should_xxx_When_xxx" . I would suggest for example "should_ReturnAllTasksCaseInsensitive_When_PerformingWildcardQuery".

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java [hunk 2]
```diff
@@ -61,6 +61,32 @@ void should_ReturnAllTasks_For_ProvidedSearchValueAndAdditionalParameters() {
...
+    WildcardSearchField[] wildcards = {WildcardSearchField.NAME};
+
+    List<TaskSummary> foundTasksSensitive =
+        taskService
+            .createTaskQuery()
+            .wildcardSearchFieldsIn(wildcards)
+            .wildcardSearchValueLike("%Wid%")
+            .list();
+
+    List<TaskSummary> foundTasksInSensitive =
```
Comment (tge20) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java:78]: Maybe more meaningful to call it "foundTasksUpperCase" and "foundTasksLowerCase".

**Review by tge20 — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java (hunk 1 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java:66]: Done

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java (hunk 2 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java]: Decided to go with foundTasksCaseInsensitive / foundTasksSensitive after consulting with Mustapha

**Review by tge20 — DISMISSED** [review]

**Review by holgerhagen — CHANGES_REQUESTED** [review]
The PR is ok. But I stumbled into the JavaDoc of the wildcard fields setter: 
  /**
   * Add the values of the wildcard search fields for exact matching to your query. Must be used in
   * combination with the wildcardSearchValueLike parameter
   *
   * @param wildcardSearchFields the values of your wildcard search fields
   * @return the query
   */
  TaskQuery wildcardSearchFieldsIn(WildcardSearchField... wildcardSearchFields);

To me speaking of values here is misleading. What do you think? And if you agree, could you please change that with this PR?

File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java [hunk 3]
```diff
@@ -632,10 +632,10 @@ TaskQuery customAttributeLike(TaskCustomField customField, String... searchArgum
   TaskQuery wildcardSearchValueLike(String wildcardSearchValue);
 
   /**
-   * Add the values of the wildcard search fields for exact matching to your query. Must be used in
-   * combination with the wildcardSearchValueLike parameter
+   * Add the task fields for which the wildcard search should be performed as an exact match to your
+   * query. Must be used in combination with the wildcardSearchValueLike parameter
    *
-   * @param wildcardSearchFields the values of your wildcard search fields
+   * @param wildcardSearchFields the task fields of your wildcard search
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java:638]:
```suggestion
   * @param wildcardSearchFields the Task fields of your wildcard search
```

File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java [hunk 4]
```diff
@@ -632,10 +632,10 @@ TaskQuery customAttributeLike(TaskCustomField customField, String... searchArgum
   TaskQuery wildcardSearchValueLike(String wildcardSearchValue);
 
   /**
-   * Add the values of the wildcard search fields for exact matching to your query. Must be used in
-   * combination with the wildcardSearchValueLike parameter
+   * Add the task fields for which the wildcard search should be performed as an exact match to your
+   * query. Must be used in combination with the wildcardSearchValueLike parameter
    *
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/api/TaskQuery.java:637]:
```suggestion
   * Add the Task fields for which the wildcard search should be performed as an exact match to your
   * query. Must be used in combination with the wildcardSearchValueLike parameter
   *
```

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java [hunk 5]
```diff
@@ -61,6 +61,32 @@ void should_ReturnAllTasks_For_ProvidedSearchValueAndAdditionalParameters() {
...
+    List<TaskSummary> foundTasksCaseInsensitive =
+        taskService
+            .createTaskQuery()
+            .wildcardSearchFieldsIn(wildcards)
+            .wildcardSearchValueLike("%wid%")
+            .list();
+
+    assertThat(foundTasksCaseSensitive).hasSize(80);
+    assertThat(foundTasksCaseInsensitive).hasSize(80);
+    assertThat(foundTasksCaseInsensitive).containsExactlyElementsOf(foundTasksCaseSensitive);
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTasksByWildcardSearchAccTest.java:87]:
```suggestion
    assertThat(foundTasksCaseInsensitive).containsExactlyElementsOf(foundTasksCaseSensitive);
```

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

**Review by holgerhagen — APPROVED** [review]

**Pull request merged**